src/session.cpp

```diff
@@ -357,6 +323,33 @@ namespace libtorrent
 		}
 	}
 
+	namespace
+	{
+		std::vector<boost::shared_ptr<plugin>> default_plugins()
+		{
+			std::vector<boost::shared_ptr<plugin>> v;
+#ifndef TORRENT_DISABLE_EXTENSIONS
+			using wrapper = session_impl::session_plugin_wrapper;
+			boost::shared_ptr<plugin> ut_pex(new wrapper(create_ut_pex_plugin));
+			boost::shared_ptr<plugin> ut_metadata(new wrapper(create_ut_metadata_plugin));
+			boost::shared_ptr<plugin> smart_ban(new wrapper(create_smart_ban_plugin));
+
+			v.push_back(ut_pex);
```

Comment:
you should either move the pointers into the vector, or push_back() the temporary that make_shared<> returns